src/status_im2/contexts/emoji_picker/utils.cljs

```diff
@@ -0,0 +1,17 @@
+(ns status-im2.contexts.emoji-picker.utils
+  (:require [clojure.string :as string]
+            [status-im2.contexts.emoji-picker.constants :as constants]
+            [status-im2.contexts.emoji-picker.data :refer [emoji-data]]))
+
+(defn search-emoji
+  [search-query]
+  (let [cleaned-query (string/lower-case (string/trim search-query))]
+    (->> (filter (fn [{:keys [label tags emoticon]}]
+                   (or (string/includes? label cleaned-query)
+                       (when emoticon
+                         (if (vector? emoticon)
+                           (some #(string/includes? (string/lower-case %) cleaned-query) emoticon)
+                           (string/includes? (string/lower-case emoticon) cleaned-query)))
+                       (some #(string/includes? % cleaned-query) tags)))
+                 emoji-data)
+         (partition-all constants/emojis-per-row))))
```

Comment:
We can write a test for this function

Reply:
or even put a high level test on the component using this behaviour testing it at the component level (status-im2.contexts.emoji-picker.view)

Reply:
I think both the unit test and component test could be helpful, but since this is a pure, utility function, the unit test is perfect to check it works well with all sorts of inputs.